bark/models/observer/observer_model_parametric.cpp

```diff
@@ -49,36 +49,36 @@ ObservedWorld ObserverModelParametric::Observe(
      observe_only_for.end(), agent_id) == observe_only_for.end()) {
     return observed_world;
   }
-
-  AddStateDeviationFrenet(observed_world.GetEgoAgent(), ego_state_deviation_dist_);
+  const auto previous_observed_world = world->GetAgents().at(agent_id)->GetSensedWorld();
+  double delta_time = 0.0f; // must only be valid when previous observed world exists
+  if(previous_observed_world) {
+      delta_time = world->GetWorldTime() - previous_observed_world->GetWorldTime();
+  }
+  AddStateDeviationFrenet(observed_world.GetEgoAgent(), ego_state_deviation_dist_,
+                        previous_observed_world, delta_time);
   for (auto& agent : observed_world.GetOtherAgents()) {
-    AddStateDeviationFrenet(agent.second, others_state_deviation_dist_);
+    AddStateDeviationFrenet(agent.second, others_state_deviation_dist_,
+                            previous_observed_world, delta_time);
   }
 
   return observed_world;
 }
 
-void ObserverModelParametric::AddStateDeviationFrenet(const AgentPtr& agent, const DistributionPtr& multi_dim_distribution) const {
-  // Get Current Frenet State of Agent
-  const Point2d pos = agent->GetCurrentPosition();
-  const auto& lane_corridor = agent->GetRoadCorridor()->GetCurrentLaneCorridor(pos);
-  if(!lane_corridor) {
-    return;
-  };
-  FrenetState current_frenet_state(agent->GetCurrentState(), lane_corridor->GetCenterLine());
-
+void ObserverModelParametric::AddStateDeviationFrenet(const AgentPtr& agent, const DistributionPtr& multi_dim_distribution,
+                         const ObservedWorldPtr& previous_observed_world, const double& delta_time) const {
   // Add sampled frenet deviation to current frenet state
-  const auto frenet_deviation = multi_dim_distribution->Sample();
-  BARK_EXPECT_TRUE(frenet_deviation.size() == 2); // Lat, Long position deviation
+  const auto state_deviation = multi_dim_distribution->Sample();
+  BARK_EXPECT_TRUE(state_deviation.size() == 4); // directional, orthogonal position deviation + velocity and angular deviation
 
-  current_frenet_state.lon += frenet_deviation[0];
-  current_frenet_state.lat += frenet_deviation[1];
-  
-  // Convert back and set dynamic agent state 
-  const auto deviated_state = FrenetStateToDynamicState(current_frenet_state, lane_corridor->GetCenterLine());
-  agent->SetCurrentState(deviated_state);
+  // Get Current Frenet State of Agent
+  auto state = agent->GetCurrentState();
+  const auto theta = state[StateDefinition::THETA_POSITION];
+  state[StateDefinition::X_POSITION] += state_deviation[0]*cos(theta) + state_deviation[1]*sin(theta);
```

Comment:
why do we need a transformation here? we can have a distribution that defines the deviations on the Cartesian coordinates

Reply:
Unless theta is used from the sampled values. so would need to update theta before and then calculate the x, y position

Reply:
Deviations in Cartesian coordiantes would change the results if we rotate the whole scene. This formulation is independent of road placement in world coordinates.

Reply:
but theta would still be sampled, why not use the sampled one?